### ISSUE TITLE
Full half precision optimizer 

### ISSUE THREAD

**Issue by eric-haibin-lin** [microsoft/DeepSpeed#447]
Hi, 

Is there an implementation of fully half-precision optimizer? I see from the GPT-3 paper:
> ... implemented an early version of the codebase, and developed the memory optimizations for fully
half-precision training.

It looks like DeepSpeed optimizer is still using mixed precision. Has anyone looked into this before? This would allow training these big models with much less memory ... Thanks!

**Comment by tjruwase** [comment]
@eric-haibin-lin Yes, you are correct, DeepSpeed's fp16 and ZeRO optimizers are using mixed precision. We currently don't have a fully half-precision optimizer.

**Comment by eric-haibin-lin** [comment]
thank you for the reply! Is there a plan to add/evaluate fully fp16 optimizer?

**Comment by tjruwase** [comment]
We don't currently have a plan for this. Yours is the first interest in this, and so we will note it. Thanks!
